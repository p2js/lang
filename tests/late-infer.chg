extern "intrinsic" fn bitcast<T, U>(v: T): U;

fn f<T>(x: T, y: T*) {}
fn main() {
  let z = bitcast(0u64);
  f(1, z);

  match bitcast(0) {
    other => {
      let x: u32 = other;
    }
  };
}
