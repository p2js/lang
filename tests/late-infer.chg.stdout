expr @ 3:22 4:0 => ()
expr @ 5:11 5:18 => extern "intrinsic" fn bitcast(...)
expr @ 5:19 5:20 => u64
expr @ 5:11 5:24 => i32*
expr @ 6:3 6:4 => fn f(...)
expr @ 6:5 6:6 => i32
expr @ 6:8 6:9 => i32*
expr @ 6:3 6:10 => ()
expr @ 8:9 8:16 => extern "intrinsic" fn bitcast(...)
expr @ 8:17 8:18 => i32
expr @ 8:9 8:19 => u32
expr @ 10:20 10:25 => u32
expr @ 9:14 12:0 => ()
expr @ 8:3 12:4 => ()
expr @ 4:11 14:1 => ()



define void @f$LT$i32$GT$(i32 %lt.0, i32* %lt.1) {
prologue:
%l.0 = alloca i32
%l.1 = alloca i32*
store i32 %lt.0, i32* %l.0
store i32* %lt.1, i32** %l.1
br label %bb.0

bb.0:
ret void

}

define void @main() {
prologue:
%l.0 = alloca i32*
%l.5 = alloca i32
%l.6 = alloca i32
br label %bb.0

bb.0:
%t.0 = alloca i64
store i64 0, i64* %t.0
%t.1 = bitcast i64* %t.0 to i32**
%l.1 = load i32*, i32** %t.1
store i32* %l.1, i32** %l.0
%l.3 = load i32*, i32** %l.0
%l.2 = call {} bitcast(void(i32, i32*)* @f$LT$i32$GT$ to {}(i32, i32*)*)(i32 1, i32* %l.3)
br label %bb.1

bb.1:
%t.2 = alloca i32
store i32 0, i32* %t.2
%t.3 = bitcast i32* %t.2 to i32*
%l.4 = load i32, i32* %t.3
br label %bb.2

bb.2:
br label %bb.3

bb.3:
store i32 %l.4, i32* %l.5
%l.7 = load i32, i32* %l.5
store i32 %l.7, i32* %l.6
br label %bb.4

bb.4:
ret void

}


